make/Main.gmk

```diff
@@ -969,20 +969,24 @@ else
 
   jdk.jdeps-gendata: java
 
-  # The ct.sym generation uses all the moduleinfos as input
-  jdk.compiler-gendata: $(GENSRC_MODULEINFO_TARGETS) $(JAVA_TARGETS)
-  # jdk.compiler-gendata needs the BUILD_JDK. If the BUILD_JDK was supplied
+  # The ct.sym generation is done in both jdk.compiler and jdk.javadoc gendata.
+  # It needs all generated java code present.
+  jdk.compiler-gendata: $(JAVA_TARGETS)
+  jdk.javadoc-gendata: $(JAVA_TARGETS)
```

Comment:
The comment talks about source code, but this is adding JAVA_TARGETS which is all compiled classes. Is the tool operating on source code or on class files?

Reply:
I think it might operate on both, but I'm not sure. But my thinking was that JAVA_TARGETS depended on GENSRC_TARGETS, which in turn would make sure that all source code was properly generated.

Am I missing something?

Reply:
This tool only uses sources, it will not use the compiled classfiles. So, running it at a point where all sources are generated should be sufficient. Most important are generated module-infos, but other sources may in some cases be significant as well.

Reply:
Thanks, in that case I think both of these should be depending on `$(GENSRC_TARGETS)` instead.

Reply:
I take that back, jdk.javadoc-gendata should depend on `$(GENSRC_TARGETS)`, jdk.compiler-gendata still needs `$(JAVA_TARGETS)` as the recipe for that target seems to actually use compiled classes.

Reply:
Uh, sorry for not being precise. I was talking of the `JavadocElementList`/`jdk.javadoc-gendata` - that does not use classfiles, only sources. `jdk.compiler-gendata`/`ct.sym` generation surely does use (all) the new JDK classfiles. Sorry for the confusion.

Reply:
@lahodaj Just to be absolutely clear: In `jdk.javadoc-gendata`, we're calling two tools: Not only `JavadocElementList` (which only requires source, not class files), but also `CreateSymbols build-javadoc-data`. Can you confirm that this too only requires sources?

(I take it that this usage is different from `CreateSymbols build-ctsym` as called in `jdk.compiler-gendata`.)

Reply:
Looking at `make/modules/jdk.javadoc/Gendata.gmk` the recipe never references `$(JDK_OUTPUTDIR)/modules/`, the output dir where the compiled classes are located. It only references the output of `$(call GetModuleSrcPath)` (and some static src dirs), so I'm pretty confident that this is correct.

Reply:
Somewhat related to this, while investigating another bug and with this PR fresh in memory, I think we are missing another dependency. 

In `make/modules/jdk.javadoc/Gendata.gmk`, we are copying the generated files into `jdk.javadoc.interim` (part of interim langtools). This makes me believe that the generated files are needed when we use `NEW_JAVADOC` from interim langtools to build the API docs for the JDK. However, there is no dependency declared from any of the `docs-*-api-javadoc` targets to `jdk.javadoc-gendata`, so if this works today, it's just by luck.

I see two possible outcomes:

1. If the generated files from `jdk.javadoc-gendata` are expected to be present and used when generating the main API docs for the JDK, then we need to add dependencies for that.
2. If the generated files aren't actually needed, we should stop copying them to avoid unnecessary work and non deterministic behavior.

Not sure if we should hijack this PR for this problem, probably better to file a separate issue. @lahodaj what do you think about this?

Reply:
Let's treat that in a separate issue. It is, after all, a separate issue. :-)

Reply:
Just for closure of my discovered followup bug, it's not actually an issue. I found where it was introduced and the "platform links" are not needed when producing our internal API docs.
https://bugs.openjdk.org/browse/JDK-8259848